unittests/platformagnostictests.py

```diff
@@ -187,7 +187,7 @@ def test_validate_dirs(self):
 
         # Using parent as builddir should fail
         with self.subTest('parent directory'):
-            self.builddir = os.path.dirname(self.builddir)
+            self.builddir = self.common_test_dir
```

Comment:
I wonder if the solution to this is maybe that the _build_dir_root should be a single base directory in $TMPDIR, and all temporary builddirs should be created inside it?

Reply:
This is what the test was originally doing, just with more steps

Reply:
I mean that this actually made me think that we should do this:

```
/tmp/meson-tests-temp-XXXX
/tmp/meson-tests-temp-XXXX/tmpYYYYY
/tmp/meson-tests-temp-XXXX/tmpZZZZZ
```

If something goes catastrophically wrong in exiting the test harness, it's only one directory to rm -rf manually. And os.path.dirname(self.builddir) would be /tmp/meson-tests-temp-XXXX which is equally under our control, rather than being /tmp (my assumption is you didn't want the `# Using parent as builddir should fail` test to suddenly start trying to configure in /tmp itself).

Reply:
No, I actually tried that initially and it works fine, the call to `os.path.basedir` works because of the assumption that the build directory is inside the source tree.

It's really unfortunate we can't use pytest, because it has "session" level fixtures that would make this trivial. The best we can do with unittest (I think) is a per module directory, so you'd still end up with:
```
/tmp/meson-tests-allplatformtests-XXXX/
/tmp/meson-tests-windowstests-XXXX/
/tmp/meson-tests-machinefiletests-XXXX/
```